From 3b54d3e643ec5ce2a7c2683ba558f4c1b0e20bbc Mon Sep 17 00:00:00 2001
From: Marek 'marx' Grac <mgrac@redhat.com>
Date: Wed, 26 Aug 2015 12:54:47 +0200
Subject: [PATCH 2/6] fence_ipmilan: Re-introduce -t / timeout option

In the new fence agent this option was replaced by --*-timeout but old
option should work as before. It sets all *-timeout to a value 'timeout'

Resolves: rhbz#1254700
---
 fence/agents/ipmilan/fence_ipmilan.py | 16 +++++++++++++++-
 tests/data/metadata/fence_idrac.xml   |  5 +++++
 tests/data/metadata/fence_ilo3.xml    |  5 +++++
 tests/data/metadata/fence_ilo4.xml    |  5 +++++
 tests/data/metadata/fence_imm.xml     |  5 +++++
 tests/data/metadata/fence_ipmilan.xml |  5 +++++
 6 files changed, 40 insertions(+), 1 deletion(-)

diff --git a/fence/agents/ipmilan/fence_ipmilan.py b/fence/agents/ipmilan/fence_ipmilan.py
index d7c91bf..3d75218 100644
--- a/fence/agents/ipmilan/fence_ipmilan.py
+++ b/fence/agents/ipmilan/fence_ipmilan.py
@@ -125,12 +125,21 @@ def define_new_opts():
 		"help" : "",
 		"order" : 1
 	}
+	all_opt["timeout"] = {
+		"getopt" : "t:",
+		"longopt" : "timeout",
+		"help" : "-t [seconds]                   Timeout (sec) for IPMI operation",
+		"required" : "0",
+		"shortdesc" : "Timeout (sec) for IPMI operation",
+		"order" : 1
+	}
 
 def main():
 	atexit.register(atexit_handler)
 
 	device_opt = ["ipaddr", "ipport", "login", "no_login", "no_password", "passwd",
-		"lanplus", "auth", "cipher", "privlvl", "sudo", "ipmitool_path", "method", "obsolete_ip"]
+		"lanplus", "auth", "cipher", "privlvl", "sudo", "ipmitool_path", "method",
+		"obsolete_ip", "timeout"]
 	define_new_opts()
 
 	if os.path.basename(sys.argv[0]) == "fence_ilo3":
@@ -148,6 +157,11 @@ def main():
 		pi["--ip"] = pi["--obsolete-ip"]
 	options = check_input(device_opt, pi)
 
+	if "--timeout" in options:
+		options["--shell-timeout"] = options["--timeout"]
+		options["--power-timeout"] = options["--timeout"]
+		options["--login-timeout"] = options["--timeout"]
+
 	docs = {}
 	docs["shortdesc"] = "Fence agent for IPMI"
 	docs["longdesc"] = "fence_ipmilan is an I/O Fencing agent\
diff --git a/tests/data/metadata/fence_idrac.xml b/tests/data/metadata/fence_idrac.xml
index 1455897..a1c864a 100644
--- a/tests/data/metadata/fence_idrac.xml
+++ b/tests/data/metadata/fence_idrac.xml
@@ -84,6 +84,11 @@
 		</content>
 		<shortdesc lang="en">Privilege level on IPMI device</shortdesc>
 	</parameter>
+	<parameter name="timeout" unique="0" required="0">
+		<getopt mixed="-t [seconds]                   Timeout (sec) for IPMI operation" />
+		<content type="string"  />
+		<shortdesc lang="en">Timeout (sec) for IPMI operation</shortdesc>
+	</parameter>
 	<parameter name="verbose" unique="0" required="0">
 		<getopt mixed="-v, --verbose" />
 		<content type="boolean"  />
diff --git a/tests/data/metadata/fence_ilo3.xml b/tests/data/metadata/fence_ilo3.xml
index 02c9183..783e342 100644
--- a/tests/data/metadata/fence_ilo3.xml
+++ b/tests/data/metadata/fence_ilo3.xml
@@ -84,6 +84,11 @@
 		</content>
 		<shortdesc lang="en">Privilege level on IPMI device</shortdesc>
 	</parameter>
+	<parameter name="timeout" unique="0" required="0">
+		<getopt mixed="-t [seconds]                   Timeout (sec) for IPMI operation" />
+		<content type="string"  />
+		<shortdesc lang="en">Timeout (sec) for IPMI operation</shortdesc>
+	</parameter>
 	<parameter name="verbose" unique="0" required="0">
 		<getopt mixed="-v, --verbose" />
 		<content type="boolean"  />
diff --git a/tests/data/metadata/fence_ilo4.xml b/tests/data/metadata/fence_ilo4.xml
index 28a77da..ef1062d 100644
--- a/tests/data/metadata/fence_ilo4.xml
+++ b/tests/data/metadata/fence_ilo4.xml
@@ -84,6 +84,11 @@
 		</content>
 		<shortdesc lang="en">Privilege level on IPMI device</shortdesc>
 	</parameter>
+	<parameter name="timeout" unique="0" required="0">
+		<getopt mixed="-t [seconds]                   Timeout (sec) for IPMI operation" />
+		<content type="string"  />
+		<shortdesc lang="en">Timeout (sec) for IPMI operation</shortdesc>
+	</parameter>
 	<parameter name="verbose" unique="0" required="0">
 		<getopt mixed="-v, --verbose" />
 		<content type="boolean"  />
diff --git a/tests/data/metadata/fence_imm.xml b/tests/data/metadata/fence_imm.xml
index 3eb0932..d39c6b4 100644
--- a/tests/data/metadata/fence_imm.xml
+++ b/tests/data/metadata/fence_imm.xml
@@ -84,6 +84,11 @@
 		</content>
 		<shortdesc lang="en">Privilege level on IPMI device</shortdesc>
 	</parameter>
+	<parameter name="timeout" unique="0" required="0">
+		<getopt mixed="-t [seconds]                   Timeout (sec) for IPMI operation" />
+		<content type="string"  />
+		<shortdesc lang="en">Timeout (sec) for IPMI operation</shortdesc>
+	</parameter>
 	<parameter name="verbose" unique="0" required="0">
 		<getopt mixed="-v, --verbose" />
 		<content type="boolean"  />
diff --git a/tests/data/metadata/fence_ipmilan.xml b/tests/data/metadata/fence_ipmilan.xml
index 34bc669..6d6f102 100644
--- a/tests/data/metadata/fence_ipmilan.xml
+++ b/tests/data/metadata/fence_ipmilan.xml
@@ -84,6 +84,11 @@
 		</content>
 		<shortdesc lang="en">Privilege level on IPMI device</shortdesc>
 	</parameter>
+	<parameter name="timeout" unique="0" required="0">
+		<getopt mixed="-t [seconds]                   Timeout (sec) for IPMI operation" />
+		<content type="string"  />
+		<shortdesc lang="en">Timeout (sec) for IPMI operation</shortdesc>
+	</parameter>
 	<parameter name="verbose" unique="0" required="0">
 		<getopt mixed="-v, --verbose" />
 		<content type="boolean"  />
-- 
1.9.3

